test/test_runner.go

```diff
@@ -86,11 +109,38 @@ type TestRunner struct {
 	testRuntime runtime.Runtime
 
 	coverageReport *runtime.CoverageReport
+
+	// logger is injected as the program logger for the script
+	// environment.
+	logger zerolog.Logger
+
+	// logCollection is a hook attached in the program logger of
+	// the script environment, in order to aggregate and expose
+	// log messages from test cases and contracts.
+	logCollection *LogCollectionHook
```

Comment:
Just for my understanding: These logs only collect what is logged in the test script itself, but not the ones logged in the blockchain, correct?

Reply:
Yes, exactly. They also collect logs from unit tests, beside the test script itself, because all this code runs in the same test script environment. For unit tests, there's no way currently, to inject implementations like we do with the `EmulatorBackend`

Reply:
But at least, the logs are visible in the test suite output :sweat_smile: Which is already a good improvement, imho.